### PR TITLE
Field control avatar

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/constants/vars.h [hunk 1]
```diff
@@ -49,10 +49,10 @@
 // Counts up every step. Wraps around at 128.
 // When wraparound occurs, the friendship of
 // every party poke gets a slight boost.
-#define VAR_FRIENDSHIP_STEP_COUNT           0x4021
+#define VAR_HAPPINESS_STEP_COUNTER          0x4021
 
 // Maybe unused?
```
Comment (PikalaxALT) [inline include/constants/vars.h]: Delete comment

File: include/wild_encounter.h [hunk 2]
```diff
@@ -45,6 +45,6 @@ void DisableWildEncounters(bool8 state);
 u8 GetUnownLetterByPersonalityLoByte(u32 personality);
 bool8 SweetScentWildEncounter(void);
 void SeedWildEncounterRng(u16 randVal);
-void ResetEncounterRateModifiers(void);
+void ResetEncounterRateModifiers(void);bool8 TryStandardWildEncounter(u32 currMetatileBehavior);
```
Comment (jiangzhengwenjz) [inline include/wild_encounter.h]: newline

File: src/field_control_avatar.c [hunk 3]
```diff
@@ -0,0 +1,1183 @@
...
+    direction = GetPlayerMovementDirection();
+    GetInFrontOfPlayerPosition(&position);
+    return GetInteractedObjectEventScript(&position, MapGridGetMetatileBehaviorAt(position.x, position.y), direction);
+}
+
+int SetCableClubWarp(void)
+{
+    struct MapPosition position;
+
+    GetPlayerMovementDirection();  //unnecessary
```
Comment (jiangzhengwenjz) [inline src/field_control_avatar.c]: style guide: space after `//`

File: src/field_control_avatar.c [hunk 4]
```diff
@@ -0,0 +1,1183 @@
...
+    GetPlayerPosition(&position);
+    MapGridGetMetatileBehaviorAt(position.x, position.y);  //unnecessary
```
Comment (jiangzhengwenjz) [inline src/field_control_avatar.c]: style guide: space after `//`

**Pull request merged**